enterprise/models.py

```diff
@@ -4285,9 +4313,24 @@ def get_all_learners(self):
                     pending_enterprise_customer_user=pending_user,
                     group=self,
                 ))
-            return members
         else:
-            return self.members.filter(is_removed=False)
+            members = self.members.filter(is_removed=False)
+            if q:
+                ecu_filter = Q(enterprise_customer_user__id__in=ecus_ids_list)
+                pecu_filter = Q(pending_enterprise_customer_user__user_email__icontains=q)
+                members = members.filter(ecu_filter | pecu_filter)
```

Comment:
You should try to check how many SQL statements this generates. I think you could add a select related to your `self.members.filter(...)` to get it down to only a single query.